presto-main/src/test/java/com/facebook/presto/sql/planner/assertions/ExpectedValueProvider.java

```diff
@@ -13,6 +13,20 @@
  */
 package com.facebook.presto.sql.planner.assertions;
 
+/**
+ * An implementation of ExpectedValueProvider<T> should hold the values and
+ * SymbolAliases needed to call T's constructor, and call
+ * SymbolAlias.toSymbol() to get actual Symbols to pass to T's constructor.
+ * Doing this ensures that changes to T's .equals() method that result in a
+ * change to T's constructor result in a compilation error.
+ *
```

Comment:
I'd suggest the following for the first para:
An implementation of ExpectedValueProvider<T> should do the following:
* Hold the values and SymbolAliases needed to call T's constructor
* Call SymbolAlias.toSymbol() to get actual Symbols to pass to T's constructor 

(The text in the last sentence needs some clarification, but I'm not sure my suggestion below is correct.)
Doing this ensures that changes to T's .equals() method will result in a change to T's constructor, which will result in a compilation error.